### PR TITLE
[GPU] Impelement ExperimentalDetectronPriorGridGenerator-6 gpu implementation

### PR DESCRIPTION
**Pull request by tgubanova-lohika**
- Implement experimental_detectron_prior_grid_generator reference kernel
- register operation in GPU
- add single layer tests
- add unit tests

### CODE REVIEW

File: src/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/experimental_detectron_prior_grid_generator.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                                {3, 4},
+                                {-180.5, -88.5, 180.5, 88.5, -128.5, -128.5, 128.5, 128.5, -92.5, -184.5, 92.5, 184.5})
+                }
+        }
+};
+
+using ov::test::subgraph::ExperimentalDetectronPriorGridGeneratorLayerTest;
+
+INSTANTIATE_TEST_SUITE_P(smoke_ExperimentalDetectronPriorGridGenerator, ExperimentalDetectronPriorGridGeneratorLayerTest,
+                         testing::Combine(
```
Comment (wilson-seok) [inline src/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/experimental_detectron_prior_grid_generator.cpp:76]: The op supports any supported floating point. Could you please add test case for fp16?
https://docs.openvino.ai/latest/openvino_docs_ops_detection_ExperimentalDetectronPriorGridGenerator_6.html

Comment (tgubanova-lohika) [reply src/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/experimental_detectron_prior_grid_generator.cpp:79]: done

Comment (wilson-seok) [reply src/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/experimental_detectron_prior_grid_generator.cpp:79]: Thanks!

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/edpgg/experimental_detectron_prior_grid_generator_kernel_ref.cpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    return DONT_USE_IF_HAVE_SOMETHING_ELSE;
+}
+
+ParamsKey ExperimentalDetectronPriorGridGeneratorKernelRef::GetSupportedKey() const {
+    ParamsKey k;
+    k.EnableInputDataType(Datatype::F16);
+    k.EnableInputDataType(Datatype::F32);
+    k.EnableOutputDataType(Datatype::F16);
+    k.EnableOutputDataType(Datatype::F32);
+    k.EnableInputLayout(DataLayout::bfyx);
```
Comment (wilson-seok) [inline src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/edpgg/experimental_detectron_prior_grid_generator_kernel_ref.cpp:50]: <html>
<body>
<p style="outline: none !important; margin: 10px 0px 0px; padding: 0px; color: rgb(23, 43, 77); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, &quot;Fira Sans&quot;, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Please apply those blocked format for input./out layout. Those are required for dGPU</p><div class="table-wrap" style="outline: none !important; margin: 0px; padding: 0px; color: rgb(23, 43, 77); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, &quot;Fira Sans&quot;, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

Prec | Layout
-- | --
int8 | fsv16, fsv32, bsv32_fsv32
fp16 | fsv16, bsv32_fsv16
fp32 | fsv16, bsv32_fsv16

</div>
</body>
</html>

Comment (wilson-seok) [reply src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/edpgg/experimental_detectron_prior_grid_generator_kernel_ref.cpp:50]: Sorry for confusion. Could you please add fsv16, fsv32, bsv32_fsv16, bsv32_fsv32, bsv16_fsv16, bsv16_fsv32 format in input/output layout? Those are required for dGPU support.

Comment (tgubanova-lohika) [reply src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/edpgg/experimental_detectron_prior_grid_generator_kernel_ref.cpp:50]: done

**Comment by wilson-seok** [comment]
Please rebase the PR to resolve issue in openvino-mac check.

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/edpgg/experimental_detectron_prior_grid_generator_kernel_ref.cpp (hunk 2 above)
Comment (tgubanova-lohika) [reply src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/edpgg/experimental_detectron_prior_grid_generator_kernel_ref.cpp:50]: as we discussed today, I've removed these layouts for now

**Review by wilson-seok — APPROVED** [review]
LGTM

**Comment by yeonbok** [comment]
Please rebase this PR

File: src/plugins/intel_gpu/src/graph/experimental_detectron_prior_grid_generator.cpp [hunk 3]
```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2018-2022 Intel Corporation
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/experimental_detectron_prior_grid_generator.cpp:1]: 2022 is enough

File: src/plugins/intel_gpu/src/plugin/ops/experimental_detectron_prior_grid_generator.cpp [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    auto &attrs = op->get_attrs();
+    auto &priors_shape = op->get_input_shape(0);
+    auto &featmap_shape = op->get_input_shape(1);
+    auto &image_shape = op->get_input_shape(2);
+    auto inputPrimitives = p.GetInputPrimitiveIDs(op);
+    inputPrimitives.resize(1); // only priors is read
+    cldnn::experimental_detectron_prior_grid_generator prim {
+            layer_type_name_ID(op),
+            inputPrimitives,
+            outLayout,
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/plugin/ops/experimental_detectron_prior_grid_generator.cpp:36]: this should not be set here but should be set from the calc_output_layout. Also primitive doesnot need to have output_layout.

Comment (tgubanova-lohika) [reply src/plugins/intel_gpu/src/plugin/ops/experimental_detectron_prior_grid_generator.cpp]: done

File: src/plugins/intel_gpu/src/graph/experimental_detectron_prior_grid_generator.cpp (hunk 3 above)
Comment (tgubanova-lohika) [reply src/plugins/intel_gpu/src/graph/experimental_detectron_prior_grid_generator.cpp]: done

**Pull request merged**